tests/nat/runtime/test_session_manager.py

```
@@ -0,0 +1,957 @@
+    @patch('nat.cli.type_registry.GlobalTypeRegistry')
+    @pytest.mark.asyncio
+    async def test_run_raises_for_per_user_workflow(self, mock_registry):
+        """Test run() raises error for per-user workflows."""
+        mock_registry.get.return_value.get_function.return_value = create_mock_function_registration(is_per_user=True)
+
+        sm = SessionManager(config=create_mock_config(),
+                            shared_builder=MockWorkflowBuilder(),
+                            entry_function=None,
+                            shared_workflow=None)
+
+        with pytest.raises(ValueError, match="Cannot use SessionManager.run\\(\\) with per-user workflows"):
```

Ruff (0.14.8)

Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)